plugin/trino-hive/src/main/java/io/trino/plugin/hive/DirectoryCallback.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.Table;
+import org.apache.hadoop.fs.Path;
+
+public interface DirectoryCallback
+{
+    DirectoryCallback NOOP = new DirectoryCallback() {
+        @Override
+        public void onTableUpdate(Table table, Path path)
```

Comment:
Not sure whether I need to specify `Path` here. the `path` is always the same as `table.storage.location`.

Reply:
do we use both?

Reply:
The current implementation uses:
- `table` to get the partition name and to check whether the table is partitioned.
- `path` is the key used in the internal cache

> do we use both?
Yes, they are both used, but `table`/`partition` encapsulate `path` as well.